components/script/dom/element.rs

```diff
@@ -3380,7 +3382,9 @@ impl Element {
             .and_then(|data| data.client_rect.as_ref())
             .and_then(|rect| rect.get().ok())
         {
-            if doc.needs_reflow().is_none() {
+            if doc.needs_reflow().map_or(true, |condition| {
+                condition == ReflowTriggerCondition::PaintPostponed
+            }) {
```

Comment:
It might be a bit more readable to use `matches!` here.

```suggestion
            if matches!(doc.needs_reflow(), None | Some(ReflowTriggerCondition::PaintPostponed)) {
```